Migration API and migration:import command (#1315)
ezsystems/developer-documentation: docs/api/public_php_api_managing_migrations.md

# Managing migrations

You can use the PHP API to manage and run [data migrations](../guide/data_migration.md).

## Getting migration information

> To list all migration files available in the selected folder (by default, `src/Migrations/Ibexa`),

Reviewer: ```suggestion
To list all migration files available in the directory defined in configuration (by default, `src/Migrations/Ibexa`),
```

> To list all migration files available in the  directory defined in configuration (by default, `src/Migrations/Ibexa`),

Reviewer 2: ```suggestion
To list all migration files available in the directory defined in configuration (by default, `src/Migrations/Ibexa`),
```

> use the `MigrationService:listMigations()` method:

Reviewer: ```suggestion
use the `MigrationService:listMigrations()` method:
```

``` php
[[= include_file('code_samples/api/migration/src/Command/MigrationCommand.php', 24, 27) =]]
```

To get a single migration file by its name, use the `MigrationService:findOneByName()` method:

``` php
[[= include_file('code_samples/api/migration/src/Command/MigrationCommand.php', 29, 30) =]]
```

## Running migration files

To run migration file(s), use either `MigrationService:executeOne()` or `MigrationService:executeAll()`:

``` php
[[= include_file('code_samples/api/migration/src/Command/MigrationCommand.php', 31, 33) =]]
```

> Both `executeOne()` and `executeAll()` take as optional parameter the User login you want to execute the migrations as.

Reviewer 2: I have problems with understanding what does this sentence mean.

> Both `executeOne()` and `executeAll()` take as optional parameter the User login you want to execute the migrations as.

Author: ```suggestion
Both `executeOne()` and `executeAll()` can take an optional parameter: the login of the User that you want to execute the migrations as.
```
